From 1ae2936d9fb61cba0605d140a12693c3e55261bf Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 3 Apr 2022 16:16:20 +0200
Subject: [PATCH 530/544] Revert "input: kb151: Add support for VBAT regulator
 that powers the keyboard"

This reverts commit bf954f6706e7493c3455a2ea410e658b345ddb43.
---
 drivers/input/keyboard/kb151.c | 23 -----------------------
 1 file changed, 23 deletions(-)

diff --git a/drivers/input/keyboard/kb151.c b/drivers/input/keyboard/kb151.c
index 17acd4b6cf92..e0c808b4209e 100644
--- a/drivers/input/keyboard/kb151.c
+++ b/drivers/input/keyboard/kb151.c
@@ -12,7 +12,6 @@
 #include <linux/module.h>
 #include <linux/delay.h>
 #include <linux/pm_wakeirq.h>
-#include <linux/regulator/consumer.h>
 
 #define KB151_CRC8_POLYNOMIAL		0x07
 
@@ -185,7 +184,6 @@ static const struct matrix_keymap_data kb151_default_keymap_data = {
 
 struct kb151 {
 	struct input_dev *input;
-	struct regulator *vbat_supply;
 	u8 crc_table[CRC8_TABLE_SIZE];
 	u8 row_shift;
 	u8 rows;
@@ -494,11 +492,6 @@ static int kb151_probe(struct i2c_client *client)
 	if (!kb151)
 		return -ENOMEM;
 
-	kb151->vbat_supply = devm_regulator_get(dev, "vbat");
-	if (IS_ERR(kb151->vbat_supply))
-		return dev_err_probe(dev, PTR_ERR(kb151->vbat_supply),
-				     "Failed to get vbat_supply\n");
-
 	i2c_set_clientdata(client, kb151);
 
 	crc8_populate_msb(kb151->crc_table, KB151_CRC8_POLYNOMIAL);
@@ -564,21 +557,6 @@ static int kb151_probe(struct i2c_client *client)
 	dev_warn(dev, "Your kernel doesn't have CONFIG_IP5XXX_POWER enabled, keyboard charger support is disabled.\n");
 #endif
 
-	ret = regulator_enable(kb151->vbat_supply);
-	if (ret) {
-		dev_err(dev, "Failed to enable keyboard vbat supply (%d)\n", ret);
-		return ret;
-	}
-
-	return 0;
-}
-
-static int kb151_remove(struct i2c_client *client)
-{
-	struct kb151 *kb151 = i2c_get_clientdata(client);
-
-	regulator_disable(kb151->vbat_supply);
-
 	return 0;
 }
 
@@ -590,7 +568,6 @@ MODULE_DEVICE_TABLE(of, kb151_of_match);
 
 static struct i2c_driver kb151_driver = {
 	.probe_new	= kb151_probe,
-	.remove		= kb151_remove,
 	.driver		= {
 		.name		= "kb151",
 		.of_match_table = kb151_of_match,
-- 
2.34.1

